lib/errors.js

```diff
@@ -29,37 +29,37 @@ class MessageError extends Error {
             [MessageError.keys.trace]: this.trace
         };
     }
-}
+};
 
-MessageError.keys = {
+module.exports.MessageError.keys = {
     error: 'x-error',
     trace: 'x-error-trace'
 };
 
-class BounceError extends MessageError {
+class BounceError extends module.exports.MessageError {
     constructor(detail, msg) {
-        const { fields: { replyText, exchange, routingKey } = {} } = msg || {};
+        const {
+            fields: {
+                exchange = '(unknown)',
+                replyText = '(unknown)',
+                routingKey = '(unknown)'
+            } = {}
+        } = msg || {};
         const ex = exchange === '' ? '(default)' : exchange;
         super(`${detail} Code: ${replyText}, Exchange: ${ex}, Routing Key: ${routingKey}`, msg);
     }
 }
 
-class UndeliverableMessageError extends BounceError {
+module.exports.BounceError = BounceError;
+
+module.exports.UndeliverableMessageError = class UndeliverableMessageError extends BounceError {
     constructor(msg) {
         super('Message undeliverable immediately:', msg);
     }
-}
+};
 
-class UnroutableMessageError extends BounceError {
+module.exports.UnroutableMessageError = class UnroutableMessageError extends BounceError {
     constructor(msg) {
         super('Message unroutable:', msg);
     }
-}
-
-module.exports = {
-    MessageError,
-    TimeoutError,
-    UnroutableMessageError,
-    UndeliverableMessageError,
-    isFatal: require('./client').isFatalError
```

Comment:
One issue was the compiler cannot read the types of exports of this format. This could be a problem when adding these checks.